src/components/editor/mirror-mark/index.js

```diff
@@ -120,6 +123,32 @@ class MirrorMark {
         } else if (cancel) {
           cancel()
         }
+      },
+      preview: function() {
```

Comment:
If you extracted this code from the squid repo, can you attribute them in the top of this file.

Reply:
line 4

Reply:
Done.